tools/sof-kernel-log-check.sh

```diff
@@ -1,38 +1,39 @@
 #!/bin/bash
 
-begin_line=${1:-1}
+begin_time="${@:-0}"
 declare err_str ignore_str project_key
 err_str="error|failed|timed out|panic|oops"
 ignore_str="error: debugfs write failed to idle -16|error: status|iteration [01]"
 project_key="sof-audio"
 
-[[ ! "$err_str" ]] && echo "Missing error keyword list" && exit 0
-# dmesg KB size buffer size
-#dmesg_config_define=$(awk -F '=' '/CONFIG_LOG_BUF_SHIFT/ {print $2;}' /boot/config-$(uname -r))
-#dmesg_buffer_size=$( echo $(( (1<<$dmesg_config_define) / 1024 )) )
-# kernel file log buffer size
-#kernel_buffer_size=$(du -k /var/log/kern.log |awk '{print $1;}')
-# now decide using which to catch the kernel log
-#[[ $kernel_buffer_size -lt $dmesg ]] && cmd="dmesg" || cmd="sed -n '$begin_line,\$p' /var/log/kern.log"
+[[ ! "$err_str" ]] && {
+    echo "Missing error keyword list"
+    exit 0
+}
 
-# confirm begin_line is number, if it is not the number, direct using dmesg
-[[ "${begin_line//[0-9]/}" ]] && begin_line=0
-[[ "$begin_line" -eq 0 ]] && cmd="dmesg" || cmd="sed -n '$begin_line,\$p' /var/log/kern.log"
-
-#echo "run $0 with parameter '$*' for check kernel message error"
+if [ "X$begin_time" == "X0" ]; then
+    cmd="dmesg"
+else
+    date -d "$begin_time" +'%F %T' > /dev/null || {
+        echo "Error parameter for date: $begin_time"
+        echo "Support date format: date +'%F %T'"
+        exit 0
+    }
+    cmd="journalctl --dmesg --no-pager --no-hostname -o short-precise --since='$begin_time'"
+fi
 
 if [ "$ignore_str" ]; then
-    err=$(eval $cmd|grep 'Call Trace' -A5 -B3)$(eval $cmd | grep $project_key | grep -E "$err_str"|grep -vE "$ignore_str")
+    err=$("$cmd" |grep 'Call Trace' -A5 -B3)$("$cmd" | grep $project_key | grep -E "$err_str"|grep -vE "$ignore_str")
```

Comment:
Please submit these cleanups first and separately so they don't noise the functional changes later.